app/presenters/tree_builder_menu_roles.rb

```diff
@@ -0,0 +1,155 @@
+class TreeBuilderMenuRoles < TreeBuilder
+  has_kids_for Hash, [:x_build_kids]
+
+  attr_reader :rpt_menu, :role_choice
+
+  def initialize(name, type, sandbox, role_choice, rpt_menu = nil)
+    @rpt_menu    = rpt_menu || sandbox[:rpt_menu]
+    @role_choice = role_choice
+
+    super(name, type, sandbox, true)
+  end
+
+  # Used for testing convenience and to satisfy need for
+  # tree data in controller
+
+  def hash_tree
+    @hash_tree ||= x_build_tree(@tree_state.x_tree(@name))
+  end
+
+  private
+
+  def set_locals_for_render
+    {
+      :tree_id       => "#{@name}box",
+      :tree_name     => @name.to_s,
+      :bs_tree       => @bs_tree,
+      :click_url     => "/report/menu_editor/",
+      :onclick       => "miqMenuEditor",
+      :cookie_prefix => "edit_",
+      :tree_state    => true,
+      :checkboxes    => false
+    }
+  end
```

Comment:
Only `:click_url` and `:onclick` should stay after `layout/tree` to `shared/tree` change.